May 2021 relnotes.md (#199)
MicrosoftDocs/quantum-docs: articles/qdk-relnotes.md

For update instructions, please refer to the [update guide](xref:microsoft.quantum.update-qdk).

*Release date: May 24th, 2021*

## QDK Version 0.17.2105.xxxxxx

> - Updated control-plane clients (.NET, Go, Python) with new swagger. See full details in [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs/pull/13002).

Reviewer: ```suggestion
```

This item missed the cutoff date and will go into the next release.

- Added a new function to the `azure-quantum` Python client to support the translation of binary optimization terms from `npz` to Azure Quantum. See full details in [QDK Python](https://github.com/microsoft/qdk-python/pull/61).

> - Published QIR oracle compiler sample. This program allows turns classical Q# functions on Boolean inputs into quantum implementations of that functions in terms of Q# operations at the level of QIR. This allows, for example, to implement quantum algorithms that are used by many quantum algorithms readily as classical functions.

Reviewer 2: ```suggestion
- Published [QIR oracle generation sample](https://github.com/microsoft/Quantum/issues/496). This program allows turns classical Q# functions on Boolean inputs into quantum implementations of that functions in terms of Q# operations at the level of QIR. This allows, for example, to implement quantum algorithms that are used by many quantum algorithms readily as classical functions.
```

- Added Chemistry azure-quantum sample notebook to Quantum repo. 
- Fixed a bug that prevents QIR generation from being enabled in the iqsharp-base Docker image. See details [here](https://github.com/microsoft/iqsharp/issues/432).

> - Implemented special functions proposal.

Reviewer 3: Can we add a link to the proposal and PR?

Author: @efratshabtai the ADO did not link to a PR. @cgranade would you mind adding?

Reviewer 3: @cgranade

Reviewer 4: ```suggestion
- Implemented new special functions (e.g.: factorial and log-gamma) in Microsoft.Quantum.Math (microsoft/QuantumLibraries#448). Thanks to @TheMagicNacho for the contribution (microsoft/QuantumLibraries#440)!
```

- C# Client: Changed input data format type to "v2" for Quantum Computing. 
- Set CLI user agent to include version number.

> ## Azure Quantum update

Reviewer 3: The service deployed every week - when were these updated?
Do we add a date to each service update?

Author: ```suggestion
## Azure Quantum update
*May 18th, 2021*

```

> 

Reviewer 3: This is a service side change, when was this deployed? It should be another entry in the release notes. (not part of the QDK)

> - Job Scheduler: Fixed an issue where jobs could be rejected if an unrelated metric has reached capacity.

Author: ```suggestion

```

- Control Plane: Set Provider in Failed state if provisioning fails. Previously it would be stuck in Launching/Updating state.

> - Added help button to direct user to support forum.

Reviewer 5: Added help button in portal to direct user to support forum.

> - Added help button to direct user to support forum.

Author: ```suggestion
*May 4, 2021*
- Added help button to direct user to support forum.
```

> - Added help button to direct user to support forum.

Author: ```suggestion
- Added help button in portal to direct user to support forum.
```

> - Rendered provider cost in localized currency from Azure Marketplace.

Reviewer 5: Added feedback button in portal to gather user feedback.

Author: ```suggestion
- Rendered provider cost in localized currency from Azure Marketplace.
- Added feedback button in portal to gather user feedback.
- Added quickstart guide in portal in overview blade.
```

> 

Reviewer 5: Added quickstart guide in portal in overview blade.

*Release date: May 24th, 2021*

## QDK Version 0.17.2105.xxxxxx

- Added a new function to the `azure-quantum` Python client to support the translation of binary optimization terms from `npz` to Azure Quantum. See full details in [QDK Python](https://github.com/microsoft/qdk-python/pull/61).
- Published [QIR oracle generation sample](https://github.com/microsoft/Quantum/issues/496). This program allows turns classical Q# functions on Boolean inputs into quantum implementations of that functions in terms of Q# operations at the level of QIR. This allows, for example, to implement quantum algorithms that are used by many quantum algorithms readily as classical functions.
- Added Chemistry azure-quantum sample notebook to Quantum repo. 
- Fixed a bug that prevents QIR generation from being enabled in the iqsharp-base Docker image. See details [here](https://github.com/microsoft/iqsharp/issues/432).
- Implemented new special functions (e.g.: factorial and log-gamma) in Microsoft.Quantum.Math (microsoft/QuantumLibraries#448). Thanks to @TheMagicNacho for the contribution (microsoft/QuantumLibraries#440)!
- C# Client: Changed input data format type to "v2" for Quantum Computing. 

> - Set CLI user agent to include version number.

Reviewer 6: ```suggestion
- Released Az CLI quantum extension version 0.5.0: Adapted to 'az' tool version 2.23.0, adding user agent information on calls to Azure Quantum Service.
```

> ## Azure Quantum update

Author: ```suggestion
## Azure Quantum service update
```

> *May 18th, 2021*

Author: ```suggestion
*May 25th, 2021*
- Added PA and SSMC solvers along with preview access via a specialized SKU available to a subset of customers.
*May 18th, 2021*
```

- Published [QIR oracle generation sample](https://github.com/microsoft/Quantum/issues/496). This program allows turns classical Q# functions on Boolean inputs into quantum implementations of that functions in terms of Q# operations at the level of QIR. This allows, for example, to implement quantum algorithms that are used by many quantum algorithms readily as classical functions.
- Added Chemistry azure-quantum sample notebook to Quantum repo. 
- Fixed a bug that prevents QIR generation from being enabled in the iqsharp-base Docker image. See details [here](https://github.com/microsoft/iqsharp/issues/432).
- Implemented new special functions (e.g.: factorial and log-gamma) in Microsoft.Quantum.Math (microsoft/QuantumLibraries#448). Thanks to @TheMagicNacho for the contribution (microsoft/QuantumLibraries#440)!
- C# Client: Changed input data format type to "v2" for Quantum Computing. 
- Set CLI user agent to include version number.

## Azure Quantum update
*May 18th, 2021*
- Added support for new regions: Japan East, Japan West, UK South, UK West

> 

Author: ```suggestion
```

- Published [QIR oracle generation sample](https://github.com/microsoft/Quantum/issues/496). This program allows turns classical Q# functions on Boolean inputs into quantum implementations of that functions in terms of Q# operations at the level of QIR. This allows, for example, to implement quantum algorithms that are used by many quantum algorithms readily as classical functions.
- Added Chemistry azure-quantum sample notebook to Quantum repo. 
- Fixed a bug that prevents QIR generation from being enabled in the iqsharp-base Docker image. See details [here](https://github.com/microsoft/iqsharp/issues/432).
- Implemented new special functions (e.g.: factorial and log-gamma) in Microsoft.Quantum.Math (microsoft/QuantumLibraries#448). Thanks to @TheMagicNacho for the contribution (microsoft/QuantumLibraries#440)!
- C# Client: Changed input data format type to "v2" for Quantum Computing. 
- Released Az CLI quantum extension version 0.5.0: Adapted to 'az' tool version 2.23.0, adding user agent information on calls to Azure Quantum Service.

## Azure Quantum service update
*May 25th, 2021*

> - Added PA and SSMC solvers along with preview access via a specialized SKU available to a subset of customers.

Author: ```suggestion
- Added PA (population annealing) and SSMC (sub-stochastic Monte Carlo) solvers along with preview access via a specialized SKU available to a subset of customers.
```
